dj/sql/functions.py

```diff
@@ -758,6 +758,18 @@ def infer_type(  # noqa: F811  # pylint: disable=function-redefined
     return ct.DoubleType()
 
 
+class PercentRank(Function):
+    """
+    Window function: returns the relative rank (i.e. percentile) of rows within a window partition
+    """
+
+    is_aggregation = True
+
+    @staticmethod
```

Comment:
this could be 
```python
@PercentRank.register
def infer_type() -> ct.DoubleType:
        return ct.DoubleType()
```

And then calling would use the error from dispatch for a consistent end-user experience with exceptions.

Reply:
@shangyian @samredai the dispatch stuff was shoved in during the rewrite of compile and such. any thoughts on its usage now that we are adding new functions? I had only converted some functions to use it. You can see some functions are overloaded by using it and so the thinking was that rather than write the type checks in the functions themselves we can just have this dispatcher be aware of our own `ColumnTypes` (whereas regular single dispatch would not manage this)

Reply:
Good point @CircArgs, let me open another PR with the change. I also realized that we didn't check to see if the Raw functionality still works with the new build/compile (in fact, the tests for them are commented out).

In terms of usage, I think the dispatch pattern makes sense, but at some point we should go and convert all the functions to use this pattern so that anyone new coming in and adding a function can just copy and paste the pattern. Well, that and we probably need docs